pkg/cluster/apply.go

```diff
@@ -176,7 +176,12 @@ func (a *Apply) handleObject(obj *unstructured.Unstructured) (string, error) {
 // preprocessObject preprocesses an object for it is applied to the cluster.
 func (a *Apply) preprocessObject(obj *unstructured.Unstructured) error {
 	dm := newDefaultManaged()
```

Comment:
dryrun could exit earlier if we early out on a.DryRun before this line instead of doing this and then checking if we needed it.